jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiRequest.java

```diff
@@ -442,48 +385,40 @@ public String changeSessionId()
         if (httpSession == null)
             throw new IllegalStateException("No session");
 
-        Session session = SessionHandler.ServletAPISession.getSession(httpSession);
+        ManagedSession session = SessionHandler.ServletAPISession.getSession(httpSession);
```

Comment:
I don't understand whats going on here, why doesn't the `SessionHandler` override the `getSession()` method on request to provide its own wrapped `Session`?

Why do we need the downcasting to ManagedSession?

Reply:
not needed